File: 20220829-unfinishedMarathoner/app/src/test/java/UnfinishedMarathonerTest.java
import org.junit.jupiter.api.*;

import java.util.*;

import static org.junit.jupiter.api.Assertions.*;

class UnfinishedMarathonerTest {
  @Test
  void spareRemoval() {
    List<String> list = new ArrayList<>();
    list.add("중복");
    list.add("중복아님");

    list.add("중복");

    System.out.println(list);

    list.remove("중복");
    assertEquals(list,"[중복아님");
  }

  @Test
  void doit() {
    UnfinishedMarathoner unfinishedMarathoner = new UnfinishedMarathoner();


    assertEquals("leo",unfinishedMarathoner.solution(
        new String[] {"leo", "kiki", "eden"},new String[] {"eden", "kiki"}));

    assertEquals("vinko",unfinishedMarathoner.solution(
        new String[] {"marina", "josipa", "nikola", "vinko", "filipa"},
        new String[] {"josipa", "filipa", "marina", "nikola"}));

  }
}

File: 20220830-unfinishedMarathoner/UnfinishedMarathoner.test.js
import { solution, compareArray, stringSort } from './UnfinishedMarathoner';

test('test', () => {
  expect(stringSort(['leo', 'kiki', 'eden'])).toStrictEqual(['eden', 'kiki', 'leo']);
  expect(stringSort(['eden', 'kiki'])).toStrictEqual(['eden', 'kiki']);

  expect(stringSort(['marina', 'josipa', 'nikola', 'vinko', 'filipa']))
    .toStrictEqual(['filipa', 'josipa', 'marina', 'nikola', 'vinko']);

  expect(stringSort(['josipa', 'filipa', 'marina', 'nikola']))
    .toStrictEqual(['filipa', 'josipa', 'marina', 'nikola']);
});

test('compare', () => {
  expect(compareArray(['eden', 'kiki', 'leo'], ['eden', 'kiki']))
    .toBe('leo');
  expect(compareArray(['filipa', 'josipa', 'marina', 'nikola', 'vinko'], ['filipa', 'josipa', 'marina', 'nikola']))
    .toBe('vinko');
});

test('solution', () => {
  expect(solution(['leo', 'kiki', 'eden'], ['eden', 'kiki']))
    .toBe('leo');
  expect(solution(
    ['marina', 'josipa', 'nikola', 'vinko', 'filipa'],
    ['josipa', 'filipa', 'marina', 'nikola'],
  ))
    .toBe('vinko');
});
